sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -640,11 +640,36 @@ object ScalaJSPluginInternal {
       loadedJSEnv := {
         val log = streams.value.log
         val env = resolvedJSEnv.value
-        val libs =
+        val deps =
           resolvedJSDependencies.value.data ++ scalaJSConfigurationLibs.value
+
+        /* Implement the behavior of commonJSName without having to burn it
+         * inside NodeJSEnv, and hence in the JSEnv API.
+         * Since this matches against NodeJSEnv specifically, it obviously
+         * breaks the OO approach, but oh well ...
+         */
```

Comment:
Ok. I guess we have a good enough understanding of what our options going forward are and the course of action you propose seems reasonable.

So just to resume what we'll do going forward:

1. Merge this PR as it is.
2. Separate all the `jsDependencies` logic into a separate plugin, making it clear that it is on life support only (that's what I meant with the "bundler plugin").
3. If the need arises, add a setting in the plugin to trigger the common JS behavior.